### PR TITLE
Feature/strand count bug

### PR DESCRIPTION
**Pull request by keiranmraine**
Initial functioning version which allows data as described in #35 to succeed.  Internal CI running.

Sorry, there was whitespace hell, so you'll want to view with whitespace changes off (append `?w=1` to url)

### CODE REVIEW

File: lib/Sanger/CGP/CavemanPostProcessor.pm [hunk 1]
```diff
@@ -63,298 +63,278 @@ my $norm_readnames_hash;
...
+  my $loc_counts = {
+    1 => {A => 0, C => 0, G => 0, T => 0},
+    -1 => {A => 0, C => 0, G => 0, T => 0},
+  };
+
+  foreach my $readnom(@$readname_arr){
+    my ($read_to_use, $read_to_use_2);
+    if(exists $hashed_reads->{$readnom}->{1} && exists $hashed_reads->{$readnom}->{-1}) {
+    # detected in both take read with the highest qscore
+      if($hashed_reads->{$readnom}->{1}->{qscore} == $hashed_reads->{$readnom}->{-1}->{qscore}) {
```
Comment (davidrajones) [inline lib/Sanger/CGP/CavemanPostProcessor.pm]: I thought defined behaviour for strand based counting was:
If strands are different, count both. If strands are the same, increment only one strand, 
```suggestion
      if($hashed_reads->{$readnom}->{1}->{qscore} == $hashed_reads->{$readnom}->{-1}->{qscore}) {
```
```suggestion
      if($hashed_reads->{$readnom}->{1}->{qscore} == $hashed_reads->{$readnom}->{-1}->{qscore}) {
```

Comment (keiranmraine) [reply lib/Sanger/CGP/CavemanPostProcessor.pm]: That was the original intent but before the extra block where you attempted to balance  +/- contribution.

My original attempt only fixed the issue with both being assigned to the +ve strand and that it only ever was set for a disagreement if reads overlapped (the reason we end up with it failing).  It had no logic at all that handled when the reads agree.

https://github.com/cancerit/cgpCaVEManPostProcessing/pull/36/commits/d73ed23e3728d2d555bc1c644c5fbe71de2c1b8a

I subsequently switched to the base qual as the second block (deleted) attempts to balance the +/-ve contribution, thus no longer double counting anything.

https://github.com/cancerit/cgpCaVEManPostProcessing/pull/36/commits/536d59cb10467a06aea19b124afd01288e4ccf21#diff-8614509bb785fd3a892d6da53278dc2bL311-L321

**Comment by keiranmraine** [comment]
FYI, something else is triggering an empty data set in the test WXS sample with the latest change

File: lib/Sanger/CGP/CavemanPostProcessor.pm (hunk 1 above)
Comment (keiranmraine) [reply lib/Sanger/CGP/CavemanPostProcessor.pm]: Still takes both reads if the qscore is the same.

**Comment by keiranmraine** [comment]
To be released with #37

**Pull request merged**